src/fiber/fiber.mli

```diff
@@ -322,6 +317,21 @@ module Stream : sig
 end
 with type 'a fiber := 'a t
 
+module Pool : sig
```

Comment:
We need some documentation here. This module is quite simple, but without doc, it has several possible interpretations. For instance, a pool could be understood as a "thread pool", and be interpreted the same as "Throttle" for instance.

However, this module is really about asynchronously submitting jobs that will be run elsewhere, in another execution context. The benefit of this is that `task` returns immediately without waiting for `f` to finish. However, `f` doesn't have access to the current execution context. So for instance:

- exceptions raised by `f` won't be caught by the currently active error handler, instead they will be caught by the error handler active at the point `run` is called
- `f` won't see the local variables defined at the point `task` is called. Instead, it will see the variable defined at the point `run` is called

All this is not completely obvious from the outside, and should be documented.